combat-trainer.lic

```diff
@@ -355,11 +355,25 @@ class LootProcess
   end
 
   def should_perform_ritual(prepare_consume)
-    !prepare_consume &&
-      ((@necro_train_first_aid && DRSkill.getxp('First Aid') < 32) || DRSkill.getxp('Thanatology') < 32)
+   !prepare_consume &&
+    ((@ritual_type == "dissect" && DRSkill.getxp('First Aid') < 32) ||
+    (@ritual_type == "harvest" && DRSkill.getxp('Skinning') < 32) ||
+    DRSkill.getxp('Thanatology') < 32)
+  end
+
+  def check_ritual_xp
+    DRSkill.getxp('First Aid') > 31 && DRSkill.getxp('Skinning') > 31
+  end
+
+  def cycle_rituals
```

Comment:
I'm not suggesting you make everyone cycle rituals. Or, rather, another way to look at the current logic is that we're already cycling between, say, 'preserve' and 'don't do a ritual' (in the case where Thanatology is locked). I think the new logic is just an extension of what we already have.